src/Generic.Repository/Models/PageAggregation/PageFilterAbstract.cs

```diff
@@ -1,281 +1,75 @@
-            Order = PageConfig.Order;
-            Size = PageConfig.Size;
-            TotalElements = Count;
-
-            TotalPage = TotalElements / Size;
-
-            if (needMap)
-            {
-                ThrowErrorIf.IsNullValue(mapperTo, nameof(mapperTo), string.Empty);
-                Content = mapperTo(GetItems()).ToList();
-            }
+            StartCtor(config, listEntities, cacheRepository, mapping);
         }
 
-        #endregion Ctor
-
-        /// <summary>Gets the content.</summary>
-        /// <value>The content of page.</value>
-        public virtual IReadOnlyList<TResult> Content { get; set; }
-
-        /// <summary>Gets the number page.</summary>
-        /// <value>The number page.</value>
-        public virtual int NumberPage { get; set; }
-
-        /// <summary>Gets the order.</summary>
-        /// <value>The order of page.</value>
-        public virtual string Order { get; set; }
-
-        /// <summary>Gets the size.</summary>
-        /// <value>The size of page.</value>
-        public virtual int Size { get; set; }
-
-        /// <summary>Gets the sort.</summary>
-        /// <value>The pagination sort.</value>
-        public virtual string Sort { get; set; }
-
-        /// <summary>Gets the total elements.</summary>
-        /// <value>The total elements in a page.</value>
-        public virtual int TotalElements { get; set; }
+        public void StartCtor(
+        IPageConfig config,
+        IQueryable<TValue> listEntities,
+        ICacheRepository cacheRepository,
+        Func<IEnumerable<object>, IEnumerable<TResult>> mapping)
+        {
+            ThrowErrorIf.IsNullValue(mapping, nameof(mapping), typeof(PageAbstract<,>).Name);
+            Mapping = mapping;
+        }
 
-        /// <summary>Gets the total page.</summary>
-        /// <value>The total page.</value>
-        public virtual int TotalPage { get; set; }
+        #endregion CTOR
 
         /// <summary>Gets the items.</summary>
         /// <returns></returns>
-        protected IReadOnlyList<TValue> GetItems()
+        protected new async Task<IReadOnlyList<TResult>> GetItems(CancellationToken token)
         {
-            var orderBy = PageConfig.CreateGenericOrderBy<TValue, TFilter>(_cacheRepository, default).GetAwaiter().GetResult();
-
-            var list = !Sort.Equals(PageSort.ASC.ToString())
-            ? ListEntities.OrderByDescending(orderBy)
-            : ListEntities.OrderBy(orderBy);
-
-            var skipNumber = NumberPage * Size;
+            var result = await GetQueryable(token).
+                ConfigureAwait(false);
 
-            var result = list.Skip(skipNumber).Take(Size);
-
-            return result.ToList();
+            return Mapping(result).ToList();
         }
     }
 
-    public abstract class PageFilterAbstract<TValue, TFilter> : PageFilterAbstract<TValue, TFilter, TValue>
+    public abstract class PageFilterAbstract<TValue, TFilter> : PageAttrAbstract<TValue, TValue>
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['TFilter' is not used in the class.](https://app.codacy.com/manual/guilhermecaixeta/Generic.Service.DotNetCore2.2/pullRequest?prid=4961311)